### ISSUE TITLE
Network mapped paths resolve differently in Atom 1.12

### ISSUE THREAD

**Issue by Ben3eeE** [atom/atom#13104]
### Prerequisites
- [X] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?
- [X] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
- [X] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
- [X] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
- [X] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
- [X] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.
### Description

Since Atom 1.12 beta the paths of files on network mapped drives are resolved with the full server name instead of the path of the network mapped drive. This caused problems with restoring the session of projects on network mapped drives when upgrading to Atom 1.12.

Example: `\\server\path\to\folder` is mapped to `Y:\`. Atom restores the session with the `Y:\` path but all new files open with the `\\server\path\to\folder` path. So you have to close and reopen all files you previously had opened to not have duplicate files open. This also can cause the UNC path error when running the linter on beta.

I did discover that the `tree-view` still has the old path if you copy the full path of some file and the title bar of Atom also displays the network mapped drive label. So currently we display both of the paths in different parts of Atom.
##### Atom 1.12.0-beta5 paths

![path stuff](https://cloud.githubusercontent.com/assets/1058982/19836364/6d5ec3dc-9e9e-11e6-955e-bc29966a3d65.gif)
##### Atom 1.11.2 paths

![path stuff 1 11](https://cloud.githubusercontent.com/assets/1058982/19836376/21a003a6-9e9f-11e6-8e65-6519716dfa3c.gif)
##### Upgrading from 1.11 to 1.12

![path stuff upgrade](https://cloud.githubusercontent.com/assets/1058982/19836383/6b5bae64-9e9f-11e6-952b-079c82ef0cc0.gif)
### Steps to Reproduce
1. Install Atom 1.11.2
2. Map a network drive on Windows
3. Open a folder on the network mapped drive
4. Open some files for example: `Y:\file.txt`
5. Install Atom 1.12-beta5
6. Paths from previous session restore as `Y:\file.txt`
7. Open `Y:\file.txt` from the `tree-view` a duplicate tab opens as `\\server\path\to\folder\file.txt`

**Expected behavior:** Paths to be consistent across Atom. Upgrade to be smooth so users don't have to close every file after upgrading to avoid opening duplicate files.

**Actual behavior:** Paths are inconsistent. Upgrade from 1.11 to 1.12 not smooth.
### Versions

```
atom --version

Atom    : 1.12.0-beta5
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0
```

```
apm --version

apm  1.12.5
npm  3.10.5
node 4.4.5
python 2.7.11
git 2.10.1.windows.1
visual studio 2015
```

**Comment by Ben3eeE** [comment]
Another thing only happening in project folders on network mapped drives in 1.12. Gif is using `1.12.0-beta5`.
![far double](https://cloud.githubusercontent.com/assets/1058982/19906143/e5d29e06-a079-11e6-821f-1f39c7d1c99e.gif)

**Comment by srt4rulez** [comment]
I'm also experiencing this, and it's breaking a few of my php linters. Also git diff wasn't working at all. 

```
Atom    : 1.12.2
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0
```

Windows 10

**Comment by damieng** [comment]
This was caused by an unintended change in nodejs as described at https://github.com/nodejs/node/pull/7559 when they changed the implementation of fs.realpath to use libuv.

It is fixed in node 6.4 by way of https://github.com/nodejs/node/pull/7899

**Comment by Ben3eeE** [comment]
Closed via https://github.com/atom/atom/pull/13247 🎉
